metadata/metadata.go

```diff
@@ -23,31 +31,66 @@ var (
 )
 
 func InitMetadata(ctx context.Context) error {
-	initK8sRKESystemImages()
-	initAddonTemplates()
-	initServiceOptions()
-	initDockerOptions()
+	data, err := loadData()
+	if err != nil {
+		return err
+	}
+	initK8sRKESystemImages(data)
+	initAddonTemplates(data)
+	initServiceOptions(data)
+	initDockerOptions(data)
 	return nil
 }
 
+// this method loads metadata from three place, if METADATA_URL is provided then load data from specified location. Otherwise load data from bindata.
+func loadData() (kdm.Data, error) {
```

Comment:
Can we add some logging and debug logging in this function to know:
- What source of data is being used
- When it is trying to retrieve data from URL or file
- That it is trying to retrieve something

Reply:
Updated